tests/test_adaptor_neighbour_list.cc

```diff
@@ -96,7 +96,7 @@ namespace rascal {
     }
     int np{0};
     for (auto atom : pair_manager) {
-      for (auto pair : atom) {
+      for (auto pair : atom.pairs()) {
         np++;
       }
```

Comment:
I just realize this now: Nothing is really tested here except iteration?

Reply:
it seems like it. At least the functionality compiles and runs.

Reply:
Could you take the opportunity of working on this to add a few basic tests?

Reply:
the adaptor is tested already. I am not sure what I could add so I would leave it for now.